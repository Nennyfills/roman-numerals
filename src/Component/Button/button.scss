.button {
  button {
    width: auto;
    display: inline-block;
    margin-right: 0.5rem;
    margin-bottom: 0.8rem;
    padding: 8px 16px;
    border-radius: 5px;
    text-align: center;
    border: none;
    cursor: pointer;
    font-weight: 600;
    letter-spacing: -0.006px;
    height: 42px;
    font-size: 16px;
    line-height: 8px;
    background-color: #3ab795;
    color: #ffffff;
    outline: none;
    &:disabled{
      color: #ffff;
      box-shadow: none;
      cursor: not-allowed;
      background: #b0e4d6;
      border-radius: 6px;
      &:hover {
        text-decoration: none;
        background: #b0e4d6;
        box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
        border-color: #b0e4d6;
      }
    }
    &:hover {
      text-decoration: none;
      background: #2a9d89;
      box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
      border-color: #2a9d89;
    }
  }
}
