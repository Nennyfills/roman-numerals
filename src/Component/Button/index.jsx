import React from 'react';
import PropTypes from 'prop-types';

import './button.scss';

const Button = ({
  name, onClick, disabled, btnId
}) => (
  <div className="button">
    <button
      type="submit"
      id={btnId}
      data-testid={btnId}
      onClick={onClick}
      disabled={disabled}
    >
      {name}
    </button>
  </div>
);

Button.defaultProps = {
  name: 'name',
  onClick: () => 'click',
  disabled: true,
  btnId: 'text',
};
Button.propTypes = {
  name: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  btnId: PropTypes.string,
};
export default Button;
