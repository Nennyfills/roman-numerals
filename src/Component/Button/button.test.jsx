import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Button from './index';

describe('Test Button component', () => {
  test('renders a message', () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot(`
      <div />
    `);
  });
  test('Test button clicked', () => {
    const { getByTestId } = render(<Button />);
    const btn = getByTestId('text');

    fireEvent.click(btn);
  });
});
