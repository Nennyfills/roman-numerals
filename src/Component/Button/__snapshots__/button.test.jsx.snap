// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component renders a message: 
      <div />
     1`] = `
<div
  class="button"
>
  <button
    data-testid="text"
    disabled=""
    id="text"
    type="submit"
  >
    name
  </button>
</div>
`;
